std/bitmanip.d

```diff
@@ -4005,3 +4005,444 @@ auto bitsSet(T)(T value) @nogc pure nothrow
     foreach (i; 0..63)
         assert(bitsSet(1UL << i).equal([i]));
 }
+
+pure @safe @nogc nothrow {
+
+private template upTo(T)
+{
+    enum upTo = T.sizeof * 8UL;
+}
+
+private template minUnsignedIntegral(T)
+{
+    static if (is(T == byte))
+    {
+        alias minUnsignedIntegral = ubyte;
+    }
+    else static if (is(T == short))
+    {
+        alias minUnsignedIntegral = ushort;
+    }
+    else static if (is(T == int))
+    {
+        alias minUnsignedIntegral = uint;
+    }
+    else static if (is(T == long))
+    {
+        alias minUnsignedIntegral = ulong;
+    }
+    else
+    {
+        alias minUnsignedIntegral = T;
+    }
+}
+
+/** Tests if the bit `idx` is set in integer `bitfield`.
+If `idx >= T.sizeof * 8` the bahaviour is undefined.
+
+Params:
+   bitfield = the integer containing bit to test
+   idx = the index of the bit to test
+
+Returns: `true` if the bit is set, `false` otherwise.
+*/
+bool testBit(T)(const T bitfield, const ulong idx) if (isIntegral!T)
+{
+    return (cast(ulong)(bitfield) & (1UL << idx)) > 0UL;
+}
+
+///
+unittest
+{
+    int a = 0b0000_0000_0010_0001;
+    assert(testBit(a, 0));
+    assert(!testBit(a, 1));
+    assert(testBit(a, 5));
+    assert(!testBit(a, 6));
+}
+
+/** Tests if any bit is set in the integer `bitfield`.
+
+Params:
+   bitfield = the integer to test
+
+Returns: `true` if any bit is set, `false otherwise.
+*/
+bool testAnyBit(T)(const T bitfield) if (isIntegral!T)
+{
+    return bitfield != T.init;
```

Comment:
Really? I have to throw the towel here. A function that tests whether any bit is set within an integral is an insult to the potential user's intelligence.